app/assets/javascripts/cart.js

```diff
@@ -0,0 +1,183 @@
+function shoppingCart(cartName) {
+    this.cartName = cartName;
+    this.items = [];
+    this.clearCart = false;
+
+    this.loadItems();
+    var self = this;
+    $(window).unload(function () {
+        if (self.clearCart) {
+            self.clearItems();
+        }
+        self.saveItems();
+        self.clearCart = false;
+    });
+}
+
+shoppingCart.prototype.loadItems = function () {
+    var items = localStorage !== null ? localStorage[this.cartName + "_items"] : null;
+    if (items !== null && JSON !== null) {
+        try {
+            items = JSON.parse(items);
+            for (var i = 0; i < items.length; i++) {
+                var item = items[i];
+                if (!item.empty) {
+                    this.items.push(item);
+                }
+            }
+        }
+        catch (err) {
+            // ignore errors while loading...
+        }
+    }
+};
+
+shoppingCart.prototype.saveItems = function () {
+    if (localStorage !== null && JSON !== null) {
+        localStorage[this.cartName + "_items"] = JSON.stringify(this.items);
+    }
+};
+
+// adds an item to the cart
+shoppingCart.prototype.addItem = function (cart-item) {
```

Comment:
Expected ')' to match '(' from line 42 and instead saw '-'.<br>'function closure expressions' is only available in Mozilla JavaScript extensions (use moz option).<br>Missing semicolon.<br>Expected an identifier and instead saw ')'.<br>Expected an assignment or function call and instead saw an expression.<br>'cart' is defined but never used.